src/Security/Http/AnonymousFollowerAuthenticationSuccessHandler.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace AppBundle\Security\Http;
+
+use AppBundle\Security\Http\Session\AnonymousFollowerSession;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Core\Authentication\Token\AnonymousToken;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Http\Authentication\DefaultAuthenticationSuccessHandler;
+use Symfony\Component\Security\Http\HttpUtils;
+
+class AnonymousFollowerAuthenticationSuccessHandler extends DefaultAuthenticationSuccessHandler
+{
+    private $anonymousFollowerSession;
+
+    public function __construct(HttpUtils $httpUtils, array $options = [], AnonymousFollowerSession $anonymousFollowerSession)
+    {
+        parent::__construct($httpUtils, $options);
+
+        $this->anonymousFollowerSession = $anonymousFollowerSession;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function onAuthenticationSuccess(Request $request, TokenInterface $token)
+    {
+        if (!$token instanceof AnonymousToken && $this->anonymousFollowerSession->isStarted()) {
```

Comment:
do we really need to check if the token is not a AnonymousToken here (in `onAuthenticationSuccess`) ?

Reply:
Yes we do. Authenticating an anonymous user still triggers the success handler.